sway/desktop/render.c

```diff
@@ -1821,6 +1970,58 @@ static void render_seatops(struct sway_output *output,
 	}
 }
 
+static void render_fullscreen_con(pixman_region32_t *damage,
+		struct sway_output *output, struct sway_container *fullscreen_con,
+		struct sway_workspace *workspace, bool clear_whole_screen) {
+	struct wlr_output *wlr_output = output->wlr_output;
+
+	bool on_focused_workspace = workspace == output->current.active_workspace;
+
+	// Only clear the transformed fullscreen bounds
+	pixman_region32_t dmg;
+	pixman_region32_init(&dmg);
+	if (!clear_whole_screen) {
+		pixman_region32_copy(&dmg, damage);
+		adjust_damage_to_workspace_bounds(&dmg, on_focused_workspace, output);
+		damage = &dmg;
+	}
+
+	float clear_color[] = {0.0f, 0.0f, 0.0f, 1.0f};
+	int nrects;
+	pixman_box32_t *rects = pixman_region32_rectangles(damage, &nrects);
+	for (int i = 0; i < nrects; ++i) {
+		scissor_output(wlr_output, &rects[i]);
+		fx_renderer_clear(clear_color);
+	}
+
+	if (fullscreen_con->view) {
+		struct decoration_data deco_data = get_undecorated_decoration_data();
+		deco_data.saturation = fullscreen_con->saturation;
+		if (!wl_list_empty(&fullscreen_con->view->saved_buffers)) {
+			render_saved_view(fullscreen_con->view, output, damage, deco_data);
+		} else if (fullscreen_con->view->surface) {
+			render_view_toplevels(fullscreen_con->view, output, damage, deco_data,
+					on_focused_workspace);
+		}
+	} else {
+		render_container(output, damage, fullscreen_con, fullscreen_con->current.focused,
+				on_focused_workspace);
+	}
+
+	for (int i = 0; i < workspace->current.floating->length; ++i) {
+		struct sway_container *floater =
+			workspace->current.floating->items[i];
+		if (container_is_transient_for(floater, fullscreen_con)) {
+			render_floating_container(output, damage, floater, on_focused_workspace);
+		}
+	}
```

Comment:
why do we render floating containers in this function? Wouldn't the fullscreen view be above floating cons?

Reply:
This is for floating containers that are transient for the fullscreened application. It's what upstream does :)